### PR TITLE
Fat 1567 : edge-inn-reach: Implement API Karate tests: Auth API - Request a valid INN-Reach Bearer token

### PR DESCRIPTION
**Pull request by nirmalr-shah**
access_token
Passed

### CODE REVIEW

File: edge-inn-reach/src/main/resources/volaris/edge-inn-reach/global/jwt-token-helper.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    * def okapitokenUser = okapitoken
+    * def okapiTenantUser = testTenant
+    * def authToken1 = 'Basic MTczNWVkMmQtYTRhMS00YWE5LWFmYmYtOWUyZjc2YzNkMTZkOjU5YjliMDc1LTVmNjAtNGRlNi04YWIxLTQyODY2M2M4OGI4ZA=='
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenUser)', 'x-okapi-tenant': '#(okapiTenantUser)', 'Authorization' : '#(authToken1)', 'x-to-code': 'fli01', 'x-from-code': '69a3d', 'Accept': 'application/json'  }
+    * configure headers = headersUser
+
+  @GetJWTToken
+  Scenario: Authenticate credentials and get a new JWT token
+    * print 'Authenticate credentials and get a new JWT token'
+    Given url 'http://localhost:8081/innreach/v2/oauth2/token'
```
Comment (gurleenkaurbp) [inline edge-inn-reach/src/main/resources/volaris/edge-inn-reach/global/jwt-token-helper.feature:18]: This URL is hardcoded and will need to be changed if ran on vagrant box, please revisit this and change it to use a dynamic URL or an important comment needs to be added why the URL is using localhost:8081

**Review by gurleenkaurbp — CHANGES_REQUESTED** [review]

File: edge-inn-reach/src/main/resources/karate-config.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    admin: {tenant: 'diku', name: 'diku_admin', password: 'admin'},
+    prototypeTenant: 'diku',
+
+    testTenant: testTenant ? testTenant: 'testTenant',
+    testAdmin: {tenant: testTenant, name: 'test-admin', password: 'admin'},
+    testUser: {tenant: testTenant, name: 'test-user', password: 'test'},
+
+    // define global features
+    login: karate.read('classpath:common/login.feature'),
+    loginRegularUser: karate.read('classpath:common/login.feature'),
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply edge-inn-reach/src/main/resources/karate-config.js:22]: they are required

File: edge-inn-reach/src/main/resources/karate-config.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      };
+
+      // Create 100 functions for uuid generation
+      var rand = function(i) {
+        karate.set("uuid"+i, function() {
+          return java.util.UUID.randomUUID() + '';
+        });
+      }
+      karate.repeat(100, rand);
+
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply edge-inn-reach/src/main/resources/karate-config.js:61]: removed unnecessary functions.

File: edge-inn-reach/src/main/resources/karate-config.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    loginRegularUser: karate.read('classpath:common/login.feature'),
+    loginAdmin: karate.read('classpath:common/login.feature'),
+    dev: karate.read('classpath:common/dev.feature'),
+    variables: karate.read('classpath:volaris/edge-inn-reach/global/variables.feature'),
+
+    globalPath: 'classpath:volaris/edge-inn-reach/global/',
+    featuresPath: 'classpath:volaris/edge-inn-reach/features/',
+    mocksPath: 'classpath:volaris/edge-inn-reach/mocks/',
+    samplesPath: 'classpath:volaris/edge-inn-reach/samples/',
+
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply edge-inn-reach/src/main/resources/karate-config.js:31]: they are required

**Review by arin-epm — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

File: edge-inn-reach/src/main/resources/volaris/edge-inn-reach/features/inn-reach-proxy.feature [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    * print 'Create JWT Token : Get Token'
+    * callonce read(globalPath + 'jwt-token-helper.feature@GetJWTToken')
+    * def responseToken = 'Bearer ' + response.access_token
+    * def headersUser = { 'Content-Type': 'application/json', 'Authorization' : '#(responseToken)', 'x-to-code': 'fli01', 'x-from-code': '69a3d', 'Accept': 'application/json'  }
+
+  Scenario: Negative scenario for JWT token
+    * print 'Negative scenario for JWT token'
+    * callonce read(globalPath + 'jwt-token-helper.feature@GetInvalidJWTToken')
+
+  Scenario: Delete central server
```
Comment (Arghya22) [inline edge-inn-reach/src/main/resources/volaris/edge-inn-reach/features/inn-reach-proxy.feature:20]: Delete central server should not be present in proxy file , It should be in transaction file , please check FAT-1569

**Review by Arghya22 — COMMENTED** [review]
Please check below points

File: edge-inn-reach/src/test/java/org/folio/EdgeInnReachApiTest.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    // default module settings
+    private static final String TEST_BASE_PATH = "classpath:volaris/edge-inn-reach/features/";
+
+    public EdgeInnReachApiTest() {
+        super(new TestIntegrationService(
+                new TestModuleConfiguration(TEST_BASE_PATH)));
+    }
+
+    @Test
+    void testGetToken() {
```
Comment (Arghya22) [inline edge-inn-reach/src/test/java/org/folio/EdgeInnReachApiTest.java:20]: Please change test name according to FAT-1569

**Review by Arghya22 — COMMENTED** [review]
Please check below points

File: edge-inn-reach/src/main/resources/volaris/edge-inn-reach/features/inn-reach-proxy.feature (hunk 5 above)
Comment (nirmalr-shah) [reply edge-inn-reach/src/main/resources/volaris/edge-inn-reach/features/inn-reach-proxy.feature]: removed

File: edge-inn-reach/src/test/java/org/folio/EdgeInnReachApiTest.java (hunk 6 above)
Comment (nirmalr-shah) [reply edge-inn-reach/src/test/java/org/folio/EdgeInnReachApiTest.java]: changed

**Review by gurleenkaurbp — APPROVED** [review]

**Review by Arghya22 — APPROVED** [review]

**Pull request merged**